src/widgets/settings_widget.h

```diff
@@ -64,6 +64,8 @@ enum GameSettingsWidgets {
 	WID_GS_RESTRICT_TYPE,      ///< Label upfront to the type drop-down box to restrict the list of settings to show
 	WID_GS_RESTRICT_DROPDOWN,  ///< The drop down box to restrict the list of settings
 	WID_GS_TYPE_DROPDOWN,      ///< The drop down box to choose client/game/company/all settings
+
+	WID_GS_SETTING_DROPDOWN = -1, ///< Dynamically created dropdown for changing setting value.
```

Comment:
does it actually need to be `-1`, or can we just do the next number?

Reply:
My understanding is:
* Positive numbers must be valid widgets. Code in widget/window.cpp will use them as indices into arrays without further bounds checking.
* Negative numbers are checked in widget/window.cpp and return early.

Examples:
* https://github.com/OpenTTD/OpenTTD/blob/master/src/window.cpp#L295
* https://github.com/OpenTTD/OpenTTD/blob/master/src/window.cpp#L726